keras/losses.py

```diff
@@ -1885,6 +1885,15 @@ def _ragged_tensor_sparse_categorical_crossentropy(y_true,
       sparse_categorical_crossentropy, from_logits=from_logits, axis=axis)
   return _ragged_tensor_apply_loss(fn, y_true, y_pred, y_pred_extra_dim=True)
 
+def _check_ground_truth_range(y_true):
+  """ Ensures ground truth labels in binary_crossentropy and 
```

Comment:
I think the requirement for binary_crossentropy and binary_focal_crossentropy is that the label should either be 0 or 1, not just within the range of 0 and 1. eg 0.5 should be treat as a invalid label.

Reply:
What about augmentations like MixUp, where the ground truth labels are smoothed as part of the transformation?

Reply:
@SmilingWolf ah, right!
@qlzh727 can you please confirm if the ground truth labels can be either 0 or 1 or anything in between 0 and 1?
Thank you

Reply:
I think you are referring to label_smoothing here. note that the check is applied to the raw label input which should be either 0 or 1. The y_true after label smoothing can be [0, 1]

Reply:
I'm not. In MixUp, the training sample is the weighted sum of two images, and the ground truth labels are themselves the weighted sums of the two original images' labels.

So, say I have image1 and and image2 with labels1 = [0, 1, 0, 1] and labels2 = [0, 1, 1, 0].
After mixing those up with a factor of 0.2, the training sample is going to be
X = (image1 * 0.2) + (image2 * (1.0 - 0.2))
and its labels are going to be
y = (labels1 * 0.2) + (labels2 * (1.0 - 0.2)) = [0.0, 1.0 , 0.8, 0.2]

This all happens within the input pipeline. For example: https://keras.io/examples/vision/mixup/#define-the-mixup-technique-function

Reply:
I see, good to know. In that sense, we might just want to limit the value to range [0, 1], and leave a comment about why we do this. The normal use case (the one leave in the docstring) is probably the one ppl will use.